src/php/ext/grpc/call.c

```diff
@@ -314,6 +314,11 @@ PHP_METHOD(Call, startBatch) {
     goto cleanup;
   }
 
+  // c-core may call rand(). If we don't call srand() here, all the
+  // random numbers being returned would be the same.
+  gpr_timespec now = gpr_now(GPR_CLOCK_REALTIME);
+  srand(now.tv_nsec);
```

Comment:
Thanks for investigating. Optional: Should we try to make it only initialize once? Or PHP client will only handle one call then completely restart?

Reply:
Latter. I tried to do this only once, and then the random number will indeed be different each run, but will stay the same on all RPCs.